src/scalar.rs

```diff
@@ -0,0 +1,87 @@
+use core::ops::Neg;
+use core::ops::{Add, AddAssign};
+use core::ops::{Mul, MulAssign};
+use core::ops::{Sub, SubAssign};
+
+use zeroize::Zeroize;
+
+/// `Scalar` Represents an integer value.
+pub struct Scalar {
+    pub(crate) inner: ark_ed_on_bls12_377::Fr,
+}
+
+// TODO: Methods to instantiate Scalar
```

Comment:
Implementing `Default` would probably be a good start since it doesn't require making any decisions about the API.